src/index.ts

```diff
@@ -1,19 +1,52 @@
-import type { App } from 'vue';
+import type { App, Component } from 'vue';
 import Stage from './components/Stage';
-import KonvaNode from './components/KonvaNode';
 import { componentPrefix } from './utils';
-import { KONVA_NODES } from './types';
+import Konva from 'konva';
+import KonvaNode from './components/KonvaNode';
+import { KonvaNodeConstructor } from './types';
 
 if (typeof window !== 'undefined' && !window.Konva) {
   require('konva');
 }
 
+
 const VueKonva = {
-  install: (app: App, options?: { prefix?: string }) => {
-    let prefixToUse = options?.prefix || componentPrefix;
+  install: (app: App, options?: { prefix?: string; customNodes?: KonvaNodeConstructor[] }) => {
+    const prefixToUse = options?.prefix || componentPrefix;
+
+    const konvaNodeConstructors: KonvaNodeConstructor[] = [
+      Konva.Arrow,
+      Konva.Arc,
+      Konva.Circle,
+      Konva.Ellipse,
+      Konva.FastLayer,
+      Konva.Image,
+      Konva.Label,
+      Konva.Line,
+      Konva.Path,
+      Konva.Rect,
+      Konva.RegularPolygon,
+      Konva.Ring,
+      Konva.Shape,
+      Konva.Sprite,
+      Konva.Stage,
+      Konva.Star,
+      Konva.Tag,
+      Konva.Text,
+      Konva.TextPath,
+      Konva.Transformer,
+      Konva.Wedge,
+      ...(options?.customNodes || []),
+    ];
 
-    [Stage, ...KONVA_NODES.map(KonvaNode)].map((k) => {
-      app.component(`${prefixToUse}${k.name}`, k);
+    const components: Component[] = [
+      Stage,
+      ...konvaNodeConstructors.map((konvaNodeConstructor) =>
+        KonvaNode(konvaNodeConstructor.name, konvaNodeConstructor),
```

Comment:
`KonvaNodeConstructor` may be renamed within code-building (mangling), which makes its `name` unreliable.

Reply:
Indeed. I will make changes so it takes a Record<string, KonvaNodeConstructor>